### DISCUSSION TITLE
Limit access to package publish permissions

### DISCUSSION THREAD

**Discussion by JesperTerkelsen** [community/community#18689] [Actions]
Continuing from here: https://github.community/t/how-to-limit-users-with-publish-privileges/140114

We need to be able to limit package push access on a repository to a smaller group of people than the people with write access.

Typically in enterprises compiling binaries and uploading binary packages is done in a controlled environment (like a CI pipeline) because there is a security vulnerability in allowing for direct package publishing.

Its an internal binary attacks, where someone malicious in a larger engineering organisation can inject backdoors into binary releases without anyone noticing because they are bypassing any code review process and just upload a package with some “extra stuff” compiled in.

The way to mitigate this is to restrict compiling + package upload to CI systems only, or a smaller group of people.

Please consider implementing a custom role where or some other way where can restrict package publishing to a smaller group of people than the people who have write admin on a repository.

**Comment by SwarajGK** [comment]
This link doesn't seem to be working anymore, have backed up the discussion anywhere?
